lib/shopify_api/plugs/admin_authenticator.ex

```diff
@@ -0,0 +1,55 @@
+defmodule ShopifyAPI.Plugs.AdminAuthenticator do
+  @moduledoc """
+  The ShopifyAPI.Plugs.AdminAuthenticator plug allows for easy admin authentication. The plug when included
+  in your route will verify Shopify signatures, that are added to the iframe call on admin page load, and
+  set a session cookie for the duration of the session.
+
+  Make sure to include the App name in the path, in our example it is included directly in the path `"/shop-admin/:app"`.
+
+  ## Example Usage
+  ```elixir
+  # Router
+  pipeline :shop_admin do
+    plug ShopifyAPI.Plugs.AdminAuthenticator
+  end
+
+  scope "/shop-admin/:app", YourAppWeb do
+    pipe_through :browser
+    pipe_through :shop_admin
+    get "/", SomeAdminController, :index
+  end
+  ```
+  """
+  import Plug.Conn
+  require Logger
+
+  alias Plug.Conn
+  alias ShopifyAPI.{AppServer, AuthTokenServer, ConnHelpers, Security, ShopServer}
+
+  @shopify_api_admin_authenticated :shopify_api_admin_authenticated
+
+  def init(opts), do: opts
+
+  def call(conn, options) do
+    if Conn.get_sesion(conn, @shopify_api_admin_authenticated) do
+      conn
+    else
+      with {:ok, app} <- ConnHelpers.fetch_shopify_app(conn),
```

Comment:
This may be worth extracting to a separate function (`do_authenticate/1`?) to keep the core `call/2` function nice and tidy.

Reply:
I like that idea!